python-minimax-nim/minimax_nim_split.py

```diff
@@ -0,0 +1,26 @@
+def minimax(state, is_maximizing):
```

Comment:
The tutorial version uses caching:

```python
from functools import cache

@cache
def minimax(state, is_maximizing):
    ...
```

Reply:
Hmm ... I don't think I'm explicit about which file the caching is added to (except for the alpha-beta one). Since this is just the solution of one of the exercises, I haven't even listed the `minimax()` function except to say that it's copied from `minimax_nim.py` and this happens before the caching is introduced in the following section.

That said, it probably makes sense to add the caching on all the files.